src/korg/fit.py

```diff
@@ -0,0 +1,41 @@
+from collections.abc import Iterable
+from typing import Callable
+
+from korg._python_interface import Array1dF64, KFloat, Linelist
+
+from ._julia_import import Korgjl
+
+type KorgFitParamsType = dict[str, KFloat] | dict[str, int] | dict[str, float]
```

Comment:
```suggestion
type KorgFitParamsType = dict[str, KFloat]
```

What you originally wrote is fairly redundant (since `KFloat` is a union that effectively includes `int` and `float`)

Reply:
With this simplification, it may be better to entirely eliminate KorgFitParamsType and just use `dict[str, KFloat]` everywhere it is needed

Reply:
done!